packages/create-cloudflare/e2e-tests/frameworks.test.ts

```diff
@@ -107,6 +107,8 @@ function getFrameworkTests(opts: {
 						input: [keys.enter],
 					},
 				],
+				unsupportedOSs: ["win32"],
+				unsupportedPms: ["yarn"],
```

Comment:
This line fails in tests: https://github.com/vitejs/vite/blob/021443c5a11eedfb176f2c65d5fab65e9c21adfb/packages/create-vite/src/index.ts#L620

Reply:
Create-vite basically tries to parse the `npm_config_user_agent` environment variable - expecting it to look like: `yarn/1.22.22 node/20.0.0` etc - to grab the version of yarn.

Because we are running create-cloudflare via `pnpx` and "faking" the package manager it doesn't work. Something (I think the bit of pnpx/npx that actually creates the child process) is stripping the version so `npm_config_user_agent` ends up just looking like: `yarn` (no version).

Reply:
Actually that doesn't work because you can't pin `yarn create` to a particlar version of the creation package.

Reply:
But it seems that the missing version from the user agent was due to code in `runFrameworkGenerator()`, which I think I have fixed.

Reply:
4c68751e03abc12143d1acb558857286f2e73ad2